engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -1368,6 +1382,131 @@ private void logBootModeParameters(Map<VirtualMachineProfile.Param, Object> para
         }
     }
 
+    /**
```

Comment:
`incrementVrResourceCount` (60 lines) and `decrementVrResourceCount` (38 lines) are big to start with and a lot of comment. Each comment could be the javadoc for a new method instead of having just these two.